test/plugins/test_lambda_release.py

```diff
@@ -0,0 +1,128 @@
+import unittest
+
+from datetime import datetime
+from mock import Mock, patch
+from cdflow_commands.plugins.aws_lambda import Release
+
+
+class TestLambdaRelease(unittest.TestCase):
+
+    @patch('cdflow_commands.plugins.aws_lambda.os')
+    @patch('cdflow_commands.plugins.aws_lambda.ZipFile')
+    def test_release_create_zips_directory(self, zip_file, mock_os):
+        config = Mock()
+        metadata = Mock()
+        boto_s3_client = Mock()
+        boto_s3_client.list_buckets.return_value = {
+            'Buckets': [],
+            'Owner': {
+                'DisplayName': 'string',
+                'ID': 'string'
+            }
+        }
+        release = Release(
+            config, boto_s3_client, 'dummy-component-name', metadata, '1.0.0'
+        )
+        release.create()
+        zip_file.assert_called_once_with('dummy-component-name.zip', 'x')
+
+    @patch('cdflow_commands.plugins.aws_lambda.os')
+    @patch('cdflow_commands.plugins.aws_lambda.ZipFile')
+    def test_release_does_not_create_bucket_if_existing(
+        self, zip_file, mock_os
+    ):
+        config = Mock()
+        metadata = Mock()
+        metadata.team = 'dummy-team-name'
+        metadata.aws_region = 'dummy-region'
+        boto_s3_client = Mock()
+        boto_s3_client.list_buckets.return_value = {
+            'Buckets': [
+                {
+                    'Name': 'mmg-lambdas-dummy-team-name',
+                    'CreationDate': datetime(2015, 1, 1)
+                },
+            ],
+            'Owner': {
+                'DisplayName': 'string',
+                'ID': 'string'
+            }
+        }
+        release = Release(
+            config, boto_s3_client, 'dummy-component-name', metadata, '1.0.0'
+        )
+        release.create()
+        assert not boto_s3_client.create_bucket.called
+
+    @patch('cdflow_commands.plugins.aws_lambda.os')
+    @patch('cdflow_commands.plugins.aws_lambda.ZipFile')
+    def test_release_creates_bucket_with_teamname(self, zip_file, mock_os):
```

Comment:
The `_with_teamname` suffix makes me think we test whether there's a teamname in the bucket name.  We do that, but I think the main aim of this test is to test whether we create bucket if it doesn't exist.  I'd consider renaming it to `test_release_creates_bucket_if_needed`